evalml/pipelines/time_series_classifcation_pipelines.py

```diff
@@ -0,0 +1,204 @@
+        proba = self.estimator.predict_proba(features.dropna(axis=0, how="any"))
+        proba.columns = self._encoder.classes_
+        return pad_with_nans(proba, max(0, features.shape[0] - proba.shape[0]))
+
+    def _compute_predictions(self, X, y, objectives):
+        """Compute predictions/probabilities based on objectives."""
+        y_predicted = None
+        y_predicted_proba = None
+        if any(o.score_needs_proba for o in objectives):
+            y_predicted_proba = self.predict_proba(X, y)
+        if any(not o.score_needs_proba for o in objectives):
+            y_predicted = self._predict(X, y, pad=True)
+        return y_predicted, y_predicted_proba
+
+    def score(self, X, y, objectives):
+        """Evaluate model performance on current and additional objectives.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): Data of shape [n_samples, n_features]
+            y (pd.Series, ww.DataColumn): True labels of length [n_samples]
+            objectives (list): Non-empty list of objectives to score on
+
+        Returns:
+            dict: Ordered dictionary of objective scores
+        """
+        X, y = self._convert_to_woodwork(X, y)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_woodwork_types_wrapper(y.to_series())
+        objectives = [get_objective(o, return_instance=True) for o in objectives]
+
+        y_encoded = self._encode_targets(y)
+        y_shifted = y_encoded.shift(-self.gap)
+        y_pred, y_pred_proba = self._compute_predictions(X, y, objectives)
+        non_nan_mask = _get_rows_without_nans(y_shifted, y_pred, y_pred_proba)
+        if y_pred is not None:
+            y_pred = y_pred.iloc[non_nan_mask]
+        if y_pred_proba is not None:
+            y_pred_proba = y_pred_proba.iloc[non_nan_mask]
+        y_labels = y_shifted.iloc[non_nan_mask]
+        return self._score_all_objectives(X, y_labels, y_pred,
+                                          y_pred_proba=y_pred_proba,
+                                          objectives=objectives)
+
+
+class TimeSeriesBinaryClassificationPipeline(TimeSeriesClassificationPipeline):
+    problem_type = ProblemTypes.TIME_SERIES_BINARY
+    _threshold = None
+
+    @property
+    def threshold(self):
+        return self._threshold
+
+    @threshold.setter
+    def threshold(self, value):
+        self._threshold = value
+
+    def _predict(self, X, y, objective=None, pad=False):
+        y_encoded = self._encode_targets(y)
+        features = self.compute_estimator_features(X, y_encoded)
+        features_no_nan = features.dropna(axis=0, how="any")
+
+        if objective is not None:
+            objective = get_objective(objective, return_instance=True)
+            if not objective.is_defined_for_problem_type(self.problem_type):
+                raise ValueError(f"Objective {objective.name} is not defined for time series binary classification.")
+
+        if self.threshold is None:
```

Comment:
This looks a bit different from `BinaryClassificationPipeline` so that I only have to call `pad_with_nans` once.